### PR TITLE
Pinning spdlog version to 1.8.5

### PR DESCRIPTION
**Pull request by cjnolet**
A new conda package for spdlog was released last night and it's causing a build issue in cuml. We've decided to pin the version to 1.8.5 in the meantime to unblock CI (and potentially for release 21.08). 

https://github.com/rapidsai/cuml/issues/4078

### CODE REVIEW

File: conda/recipes/versions.yaml
Comment (jakirkham) [inline conda/recipes/versions.yaml:152]: What about allowing patch fixes?

```suggestion
  - '>=1.8.5,<1.9'
```

Comment (cjnolet) [reply conda/recipes/versions.yaml:152]: Funny, that's what I had at first but ended up being conservative just since we know 1.8.5 works. I'm not at all opposed to changing it, though.

Comment (jakirkham) [reply conda/recipes/versions.yaml:152]: Would suggest to keep things a bit looser as very tight pinnings tend to cause issues of their own (not getting needed bug fixes without significant effort for example)

**Review by jakirkham — APPROVED** [review]

**Comment by dantegd** [comment]
@rapidsai/ops-codeowners the PR in RMM has been merged, can this PR be approved/merged to unblock cuML?

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**
